### PR TITLE
 store crud

### PR DESCRIPTION
**Pull request by jayesh-parmar**

### CODE REVIEW

File: app/Http/Controllers/StoreController.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public function store(StoreRequest $request)
+    {
+        Store::create($request->validated());
+
+        return redirect()->route('store.index')->with('success', 'store Added successfully.');
+    }
+
+    public function edit(string $storeId)
+    {
+        $store = Store::select('id', 'name')->find($storeId);
```
Comment (rajan-parmar) [inline app/Http/Controllers/StoreController.php]: now onwards to give pr with test. then I will check this PR.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/StoreController.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public function add()
+    {
+        return view('admin.pages.store.form');
+    }
+
+    public function store(StoreRequest $request)
+    {
+        Store::create($request->validated());
+
+        return redirect()->route('store.index')->with('success', 'store Added successfully.');
```
Comment (rajan-parmar) [inline app/Http/Controllers/StoreController.php]:
```suggestion
        return redirect()->route('store.index')->with('success', 'Store added successfully.');
```

File: app/Models/Store.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Concerns\HasUuids;
+
+class Store extends Model
+{
+    use HasFactory, HasUuids;
+
+    public $timestamps = false;
```
Comment (rajan-parmar) [inline app/Models/Store.php]: is required

File: database/migrations/2023_10_06_043604_create_stores_table.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('stores', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('name')->unique(); 
+        });
```
Comment (rajan-parmar) [inline database/migrations/2023_10_06_043604_create_stores_table.php]: timestamp()

File: resources/views/admin/layouts/sideheader.blade.php [hunk 5]
```diff
@@ -39,6 +39,14 @@
                     <span class="ml-4">Sizes</span>
                 </a>
             </li>
+            <li class="relative px-6 py-3">
+                <a class="inline-flex items-center w-full text-sm font-semibold transition-colors duration-150 hover:text-gray-800 dark:hover:text-gray-200" href="{{route('store.index')}}">
+                    <svg class="w-5 h-5" aria-hidden="true" fill="none" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" viewBox="0 0 24 24" stroke="currentColor">
+                        <path d="M19 11H5m14 0a2 2 0 012 2v6a2 2 0 01-2 2H5a2 2 0 01-2-2v-6a2 2 0 012-2m14 0V9a2 2 0 00-2-2M5 11V9a2 2 0 012-2m0 0V5a2 2 0 012-2h6a2 2 0 012 2v2M7 7h10"></path>
+                    </svg>
+                    <span class="ml-4">Store</span>
```
Comment (rajan-parmar) [inline resources/views/admin/layouts/sideheader.blade.php]:
```suggestion
                    <span class="ml-4">Stores</span>
```

File: resources/views/admin/pages/store/form.blade.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
+@extends('admin.layouts.main')
+@section('content')
+    <h4 class="mb-4 text-lg font-semibold text-gray-600 dark:text-gray-300">
+        @if (isset($stores)) Update @else Add @endif Store
```
Comment (rajan-parmar) [inline resources/views/admin/pages/store/form.blade.php]: $stores?

File: resources/views/admin/pages/store/index.blade.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                                    <svg class="w-5 h-5" aria-hidden="true" fill="currentColor" viewBox="0 0 20 20">
+                                        <path d="M13.586 3.586a2 2 0 112.828 2.828l-.793.793-2.828-2.828.793-.793zM11.379 5.793L3 14.172V17h2.828l8.38-8.379-2.83-2.828z"></path>
+                                    </svg>
+                                </a>
+                            </button>
+                        </div>
+                    </td>
+                    <td class="px-4 py-3">
+                        <div class="flex items-center space-x-4 text-sm">
+                            <button class="flex items-center justify-between px-2 py-2 text-sm font-medium leading-5 text-purple-600 rounded-lg dark:text-gray-400 focus:outline-none focus:shadow-outline-gray" aria-label="Edit">
```
Comment (rajan-parmar) [inline resources/views/admin/pages/store/index.blade.php]:
```suggestion
                            <button class="flex items-center justify-between px-2 py-2 text-sm font-medium leading-5 text-purple-600 rounded-lg dark:text-gray-400 focus:outline-none focus:shadow-outline-gray" aria-label="Delete">
```

File: tests/Feature/StoreTest.php [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    ])
+        ->assertRedirect(route('store.index'));
+
+    $this->assertDatabaseHas('stores', [
+        'id' => $store->id,
+        'name' => 'update 2022',
+    ]);
+});
+
+it('user can delete a store', function () {
```
Comment (rajan-parmar) [inline tests/Feature/StoreTest.php:38]: add test `user can not visit the stores page without login`

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: resources/views/admin/pages/store/form.blade.php (hunk 6 above)
Comment (jayesh-parmar) [reply resources/views/admin/pages/store/form.blade.php]: update vakhte Update store print karava mate

Comment (rajan-parmar) [reply resources/views/admin/pages/store/form.blade.php]: pan `$stores` ave che kya thi?

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: tests/Feature/StoreTest.php (hunk 8 above)
Comment (jayesh-parmar) [reply tests/Feature/StoreTest.php:38]: return view('admin.pages.store.form', ['store' => $store]); sorry sir store ave che

**Review by rajan-parmar — APPROVED** [review]

**Pull request merged**